### PR TITLE
[DOCS] Adds classification AUC ROC metric description

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR expands the DFA evaluation conceptual docs with the description of classification AUC ROC.

### Preview

[AUC ROC](https://stack-docs_1365.docs-preview.app.elstc.co/guide/en/machine-learning/master/ml-dfanalytics-evaluate.html#ml-dfanalytics-class-aucroc)

### CODE REVIEW

**Review by przemekwitek — APPROVED** [review]
LGTM

File: docs/en/stack/ml/df-analytics/ml-dfanalytics-evaluate.asciidoc [hunk 1]
```diff
@@ -163,3 +164,21 @@ The matrix contains the actual labels on the left side while the predicted
...
+=== Area under the curve of receiver operating characteristic (AUC ROC)
+
+The receiver operating characteristic (ROC) curve is a plot that represents the 
+performance of the classification process at different predicted probability 
+thresholds. It compares the true positive rate for a specific class against the 
+rate of all the other classes combined ("one versus all" strategy) at the 
+different threshold levels to create the curve.
+
+Let's see an example. You have three classes: `A`, `B`, and `C`, you want to 
+calculate AUC ROC for `A`. In this case, the number of correctly classified `A`s
```
Comment (lcawl) [inline docs/en/stack/ml/df-analytics/ml-dfanalytics-evaluate.asciidoc:179]: I think you need to use "unconstrained" backticks to make this format properly:
```suggestion
calculate AUC ROC for `A`. In this case, the number of correctly classified ``A``s 
```

File: docs/en/stack/ml/df-analytics/ml-dfanalytics-evaluate.asciidoc [hunk 2]
```diff
@@ -163,3 +164,21 @@ The matrix contains the actual labels on the left side while the predicted
...
+
+The receiver operating characteristic (ROC) curve is a plot that represents the 
+performance of the classification process at different predicted probability 
+thresholds. It compares the true positive rate for a specific class against the 
+rate of all the other classes combined ("one versus all" strategy) at the 
+different threshold levels to create the curve.
+
+Let's see an example. You have three classes: `A`, `B`, and `C`, you want to 
+calculate AUC ROC for `A`. In this case, the number of correctly classified `A`s 
+(true positives) are compared to the number of `B`s and `C`s that are
```
Comment (lcawl) [inline docs/en/stack/ml/df-analytics/ml-dfanalytics-evaluate.asciidoc:180]: Ditto re backticks:
```suggestion
(true positives) are compared to the number of ``B``s and ``C``s that are 
```

File: docs/en/stack/ml/df-analytics/ml-dfanalytics-evaluate.asciidoc [hunk 3]
```diff
@@ -163,3 +164,21 @@ The matrix contains the actual labels on the left side while the predicted
...
+The receiver operating characteristic (ROC) curve is a plot that represents the 
+performance of the classification process at different predicted probability 
+thresholds. It compares the true positive rate for a specific class against the 
+rate of all the other classes combined ("one versus all" strategy) at the 
+different threshold levels to create the curve.
+
+Let's see an example. You have three classes: `A`, `B`, and `C`, you want to 
+calculate AUC ROC for `A`. In this case, the number of correctly classified `A`s 
+(true positives) are compared to the number of `B`s and `C`s that are 
+misclassified as `A`s (false positives).
```
Comment (lcawl) [inline docs/en/stack/ml/df-analytics/ml-dfanalytics-evaluate.asciidoc:181]: Ditto:
```suggestion
misclassified as ``A``s (false positives).
```

File: docs/en/stack/ml/df-analytics/ml-dfanalytics-evaluate.asciidoc [hunk 4]
```diff
@@ -163,3 +164,21 @@ The matrix contains the actual labels on the left side while the predicted
...
+
+From this plot, you can compute the area under the curve (AUC) value, which is a 
+number between 0 and 1. The closer to 1, the better the algorithm performance.
```
Comment (lcawl) [inline docs/en/stack/ml/df-analytics/ml-dfanalytics-evaluate.asciidoc:184]: > The close to 1, the better the algorithm performance.

I got to this point and wasn't sure which "algorithm" this sentence was referring to or what aspect of its performance we were judging.  Can we be more specific  in summing this up?  For example, something like this:

> The higher the AUC, the better the model is at predicting As as As, in this case.

**Pull request merged**